app/src/main/java/com/swent/assos/ui/screens/QrCode.kt

```diff
@@ -1,9 +1,143 @@
 package com.swent.assos.ui.screens
 
+import androidx.compose.animation.AnimatedVisibility
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Card
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Surface
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.Font
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.swent.assos.R
+import com.swent.assos.model.data.Event
+import com.swent.assos.model.view.EventViewModel
+import com.swent.assos.ui.theme.Purple40
+import com.swent.assos.ui.theme.Purple80
 
 @Composable
 fun QrCode() {
-  Text("QrCode")
+  val eventViewModel: EventViewModel = hiltViewModel()
+  val events by eventViewModel.allEvents.collectAsState()
+  Scaffold(
+    topBar = {
+      Column(modifier = Modifier.fillMaxWidth()) {
+        Row(
+          modifier =
+          Modifier
+            .padding(8.dp)
+            .align(Alignment.CenterHorizontally)
+            .wrapContentHeight(align = Alignment.CenterVertically)) {
+          Text(
+            text = "Student",
+            style =
+            TextStyle(
+              fontSize = 35.sp,
+              fontFamily = FontFamily(Font(R.font.impact)),
+              fontWeight = FontWeight(400),
+              color = Purple80))
+          Text(
+            text = "Sphere",
+            style =
+            TextStyle(
+              fontSize = 35.sp,
+              fontFamily = FontFamily(Font(R.font.impact)),
+              fontWeight = FontWeight(400),
+              color = Purple40))
+        }
+      }
+    }) { paddingValues ->
```

Comment:
I believe it would be easier to visualize the content of the screen if you divide your work into multiple composables. Example : One composable for the text "StudentSphere". Then, you can call this composable in the QrCode composable. Moreover, as the title "StudentSphere" will surely be useful in other screens. If we create a separate composable for it, we won't have to write the code of the title again! We can simply call the composable function.